lib/eventFactory.js

```diff
@@ -411,6 +379,49 @@ function updateWorkflowGraph(config) {
     return Promise.resolve(workflowGraph);
 }
 
+/**
+ * Standardize event parameters
+ * @method updateWorkflowGraph
+ * @param  {Object}         config
+ * @param  {Pipeline}       config.pipeline                      Pipeline
+ * @param  {Object}         [config.pipeline.parameters]         Default build parameters
+ * @param  {Object}         config.eventConfig
+ * @param  {Object}         [config.eventConfig.meta.parameters] Customized build parameters
+ * @resolves {Object}                                            Resolves with standardized parameters
+ */
+function updateEventParameters(config) {
+    const { pipeline, eventConfig } = config;
+    const allowedParameters = Object.create(null);
+
+    // parameters not enabled
+    if (!pipeline.parameters) {
+        return Promise.resolve(null);
+    }
+
+    Object.entries(pipeline.parameters).forEach(([name, val]) => {
+        let defaultParameterValue = val;
+
+        if (typeof val === 'object') {
+            defaultParameterValue = val.value;
+        }
+
+        allowedParameters[name] = { value: defaultParameterValue };
+        if (eventConfig.meta.parameters) {
+            let actualParameterValue = eventConfig.meta.parameters[name];
+
+            if (typeof actualParameterValue === 'object') {
+                actualParameterValue = actualParameterValue.value;
+            }
+
+            if (actualParameterValue !== undefined) {
+                allowedParameters[name] = { value: actualParameterValue };
+            }
+        }
+    });
+
+    return Promise.resolve(allowedParameters);
```

Comment:
why return promise here ?

Reply:
yeah we don't need it, I will change that